### PR TITLE
chore: Update to `ruma/ruma`

### PR DESCRIPTION
**Pull request by Hywan**
It does what's on the tin.

### CODE REVIEW

File: crates/matrix-sdk-base/src/rooms/mod.rs [hunk 1]
```diff
@@ -195,12 +195,11 @@ impl BaseRoomInfo {
                 let mut o_ev = o_ev.clone();
                 o_ev.content.set_created_ts_if_none(o_ev.origin_server_ts);
 
-                let Some(owned_user_id) = get_user_id_for_state_key(m.state_key()) else {
```
Comment (bnjbvr) [inline crates/matrix-sdk-base/src/rooms/mod.rs:198]: this changes behavior, are you sure this new behavior is correct? does Ruma do the complicated parsing now? cc @toger5

Comment (Hywan) [reply crates/matrix-sdk-base/src/rooms/mod.rs:198]: Yes it does, see https://github.com/ruma/ruma/pull/1889.

Comment (bnjbvr) [reply crates/matrix-sdk-base/src/rooms/mod.rs:198]: Cool cool

**Review by bnjbvr — APPROVED** [review]
lgtm

**Comment by toger5** [comment]
Thanks for the work on this!

**Pull request merged**